android/src/app/organicmaps/car/screens/PlaceScreen.java

```diff
@@ -0,0 +1,202 @@
+    builder.setTitle(mMapObject.getTitle());
+    if (!mMapObject.getSubtitle().isEmpty())
+      builder.addText(mMapObject.getSubtitle());
+    String address = mMapObject.getAddress();
+    if (address.isEmpty())
+      address = Framework.nativeGetAddress(mMapObject.getLat(), mMapObject.getLon());
+    if (!address.isEmpty())
+      builder.addText(address);
+    return builder.build();
+  }
+
+  private void createPaneActions(@NonNull Pane.Builder builder)
+  {
+    final String phoneNumber = UiHelpers.getPhoneNumber(mMapObject);
+    if (!phoneNumber.isEmpty())
+    {
+      final Action.Builder openDialBuilder = new Action.Builder();
+      openDialBuilder.setIcon(new CarIcon.Builder(IconCompat.createWithResource(getCarContext(), R.drawable.ic_phone)).build());
+      openDialBuilder.setOnClickListener(() -> getCarContext().startCarApp(new Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + phoneNumber))));
+      builder.addAction(openDialBuilder.build());
+    }
+
+    final Action.Builder startRouteBuilder = new Action.Builder();
+    startRouteBuilder.setBackgroundColor(CarColor.GREEN);
+    startRouteBuilder.setFlags(Action.FLAG_PRIMARY);
+    startRouteBuilder.setTitle(getCarContext().getString(R.string.p2p_to_here));
+    startRouteBuilder.setIcon(new CarIcon.Builder(IconCompat.createWithResource(getCarContext(), R.drawable.ic_route_to)).build());
+    startRouteBuilder.setOnClickListener(() -> {
+      /* TODO (AndrewShkrob): Will be implemented with route planning */
+    });
+    builder.addAction(startRouteBuilder.build());
+  }
+
+  @NonNull
+  private Action createBookmarkAction()
+  {
+    final Action.Builder builder = new Action.Builder();
+    final CarIcon.Builder iconBuilder = new CarIcon.Builder(IconCompat.createWithResource(getCarContext(), R.drawable.ic_bookmarks));
+    if (mMapObject.getMapObjectType() == MapObject.BOOKMARK)
+      iconBuilder.setTint(CarColor.YELLOW);
+    builder.setIcon(iconBuilder.build());
+    builder.setOnClickListener(() -> {
+      if (MapObject.isOfType(MapObject.BOOKMARK, mMapObject))
+        Framework.nativeDeleteBookmarkFromMapObject();
+      else
+        BookmarkManager.INSTANCE.addNewBookmark(mMapObject.getLat(), mMapObject.getLon());
+    });
+    return builder.build();
+  }
+
+  @Override
+  public void onBackPressed()
+  {
+    Framework.nativeDeactivatePopup();
+  }
+
+  /** Called when MapObject is updated but FeatureId stays unchanged.
+   *  For example when MapObject is added or removed from bookmarks
+   *
+   * @param newMapObject updated MapObject with same FeatureId
+   */
+  public void updateMapObject(@NonNull MapObject newMapObject)
+  {
+    if (!mMapObject.getFeatureId().equals(newMapObject.getFeatureId()))
+    {
+      Logger.e(TAG, newMapObject.getFeatureId() + " != " + mMapObject.getFeatureId());
+      return;
+    }
+    mMapObject = newMapObject;
+    invalidate();
+  }
+
+  /**
+   * A builder of {@link PlaceScreen}.
+   */
+  public static final class Builder
```

Comment:
Is this builder really needed? It has just two options = two constructors of PlaceScreen.